File: Application/Dto/ApiResponse.cs
using IbraHabra.NET.Domain.Contract;

namespace IbraHabra.NET.Application.Dto;

public class ApiResponse<T>
{
    public T? Data { get; set; }
    public List<ApiError>? Error { get; set; }
    public required ResponseMeta Meta { get; set; }
}
public class ApiResponse
{
    public object? Data { get; set; }
    public List<ApiError>? Error { get; set; }
    public required ResponseMeta Meta { get; set; }
}
public class ResponseMeta
{
    public required string RequestId { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    public string? Version { get; set; }
}

File: Ui/src/types/ApiResponse.ts
export type ApiResponse<T> = {
  data: T
  error: ApiError[]
  meta: ResponseMeta
}
export type ResponseMeta = {
  requestId: string
  timeStamp: Date
  version?: string
}

export type ApiError = {
  code: string
  message: string
  type: number
  field?: string
}
